### ISSUE TITLE
ReplicationPad*d should accept 0-dim batch size inputs.

### ISSUE THREAD

**Issue by v0dro** [pytorch/pytorch#39127]
## 🚀 Feature
All variants of `ReplicationPad` should accept 0-dim batch size tensors for both forward and backward autograd. Related to https://github.com/pytorch/pytorch/issues/38115

Test code: 
``` python 
def test_ReplicationPad1d_Forward():
    z = torch.ones(0,3,10)
    m = torch.nn.ReplicationPad1d(3)
    m(z)

def test_ReplicationPad1d_Backward():
    z = torch.ones(0,3,10)
    m = torch.nn.ReplicationPad1d(3)
    m(z).sum().backward()

def test_ReplicationPad2d_Forward():
    z = torch.ones(0,3,10,10)
    m = torch.nn.ReplicationPad2d(3)
    m(z)

def test_ReplicationPad2d_Backward():
    z = torch.ones(0,3,10,10)
    m = torch.nn.ReplicationPad2d(3)
    m(z).sum().backward()

def test_ReplicationPad3d_Forward():
    z = torch.ones(0,3,10,10,10)
    m = torch.nn.ReplicationPad3d(3)
    m(z)

def test_ReplicationPad3d_Backward():
    z = torch.ones(0,3,10,10,10)
    m = torch.nn.ReplicationPad3d(3)
    m(z).sum().backward()
```

**Comment by rgommers** [comment]
This is duplicate with `gh-12013` (avoiding cross-linking to keep that issue as clean as possible), I'll close it.
